src/components/JitsiMeeting.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useRef, useState } from "react";
+import { init } from "..";
+import { IJitsiMeetingProps } from "../types";
+import { getRoomName } from "../utils";
+
+/**
+ * Returns the JitsiMeeting Component with access to a custom External API
+ * to be used as-it-is in React projects
+ *
+ * @param {IJitsiMeetingProps} props
+ * @example
+  ```js
+      <JitsiMeeting
+        id="testId"
+        domain="meet.jit.si"
+        options={{
+          roomName: 'TestingJitsiMeetingComponent'
+          width: '100%'
+          height: '500px'
+        }}
+        spinner={() => <>loading jitsi meeting...</>}
+        onApiReady={(externalApi) => console.log(externalApi)}
+      />
+  ```
+ */
+const JitsiMeeting = ({
+  id,
+  domain,
+  options,
+  tenant,
+  spinner,
+  onApiReady
+}: IJitsiMeetingProps) => {
+  const componentId = `${id}-jitsiMeeting`;
+  const apiRef = useRef<any>();
+  const meetingRef = useRef<any>(null);
+  const [loading, setLoading] = useState(true);
+  const [jitsiClass, setJitsiClass] = useState(null);
+
+  useEffect(() => {
+    init(domain, (err: Error|null, JitsiClass?: any): void => {
+      if (err) {
+        /* eslint-disable-next-line */
+        console.error(err);
+        return;
+      }
+      setJitsiClass(JitsiClass);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (jitsiClass) {
+      loadIFrame(jitsiClass);
+    }
+  }, [jitsiClass]);
+
+
+  const loadIFrame = (JitsiMeetExternalAPI: any) => {
```

Comment:
we should wrap this in a `useCallback`

Reply:
Given the nature of this component, I'm not entirely sure it will help, but rather make the first render take longer than needed, without getting any of the caching benefits later on. I might be wrong though, can you please explain why should I do this?

Reply:
You are right, we probably over do it on other projects for consistency with actual callbacks and for the sake of being explicit about when a function should be redeclared. I'm not sure I agree about the rendering taking longer though.